.github/workflows/ci.yaml

```diff
@@ -0,0 +1,175 @@
+name: Python code validation and build
+
+on:
+  push:
+
+jobs:
+  python-lint:
```

Comment:
should we run the linter with make lint? instead of adding the ruff action?

Reply:
I did not want to add new local dependency, co I went with pure action. But if there is a value in it, then for sure, I can add that.

Reply:
Yeah, I like that option and I will add make link in a separate follow-up PR